config/config.go

```diff
@@ -1,18 +1,275 @@
 package config
 
-import "github.com/sasswart/gin-in-a-can/openapi"
+import (
+	"flag"
+	"fmt"
+	"github.com/spf13/viper"
+	"os"
+	"path/filepath"
+)
 
-type Config struct {
-	Generator        GeneratorConfig
-	OpenAPI          openapi.Config
-	OutputPath       string
-	WorkingDirectory string
-	ConfigFilePath   string
+// SemVer should be updated on any new release!!
+const SemVer = "0.0.6"
+
+// TODO set default template path to wd/templates if not set in config
+
+// r represents System level settings for flags and environmental context tracking
+var (
+	Debug          bool
+	Dryrun         bool
+	VersionFlagSet bool
+	// ConfigFilePath is `.` if not set through the `-configFile` flag
+	ConfigFilePath string
+
+	// ProcWorkingDir is set through calling os.Getwd()
+	ProcWorkingDir string
+	ExePath        string
+)
+
+// Data represents the config data used in the day-to-day running of can
+//
+//	TODO this may be vague in definition for the sake of its legibility in use
+//	TODO check for redundancy
+type Data struct {
+	Generator
+	Template
+
+	// left public due to it's need to be unmarshalled by Data.Load()
+	TemplatesDir string `yaml:"templatesDir"`
+
+	// OpenAPIFile represents the path to the yaml OpenAPI 3 file to render
+	OpenAPIFile    string
+	absOpenAPIPath string
+
+	OutputPath    string
+	absOutputPath string
+}
+
+type Generator struct {
+	ModuleName string
+
+	// BasePackageName represents the
+	BasePackageName string
+}
+
+// Template is populated based on it's Name variable set as a CLI flag
+type Template struct {
+	Name string
+
+	// Directory should be ./templates/${Name} by default
+	Directory    string
+	absDirectory string
+}
+
+func (d *Data) Load() (err error) {
+	// Setup config pre-unmarshalling. This assumes we don't change directory before this is executed
+	ProcWorkingDir, err = os.Getwd()
+	if err != nil {
+		return fmt.Errorf("Config.load:: could not determine working directory: %w\n", err)
```

Comment:
See this kind of `Config.load` syntax re my comments in main. `fmt.Printf(err.Error())` is not specific enough, but this is well done.